### ISSUE TITLE
ZETA method to replace PSTH

### ISSUE THREAD

**Issue by KatharineShapcott** [esi-neuroscience/syncopy#169]
As far as I know doesn't have a PSTH method yet for spike events. My suggestion is not to bother and implement this cool looking method called Zeta IFR instead: https://elifesciences.org/articles/71969

Here's the kind of sales pitch figure from the paper comparing the two, as you can see it's a pretty simple method but would give nice statistics about responsivity of any event like data:
![elife-71969-fig1-v3](https://user-images.githubusercontent.com/65502584/144851222-dd6640ec-71ad-40cc-8abc-2cfd143ea94b.jpg)

There's no assumptions about the distribution of the underlying data and no need to pick bin sizes or any other parameters. 

They do have python code (https://github.com/JorritMontijn/zetapy/blob/master/main.py) but it's not up to date and is literally a line by line translation of the matlab code (https://github.com/JorritMontijn/ZETA/blob/master/getZeta.m). 

I would be happy to help implement this because I think it will be a super helpful first pass analysis for our data. Do you think it's a good fit for Syncopy?

**Comment by tensionhead** [comment]
Without having much expert knowledge about the analysis problems you face, the ZETA procedure itself looks sufficiently novel, useful and robust to me. I wonder what is the connection to the KS-Test, which also is parameter free and works on the cumulative distributions as well? But in any case from my side, please go ahead @KatharineShapcott ! 
As we don't actually have started with the *time-lock* submodul of SyNCoPy yet, you could probably best provide a **backend script**, like we already have for example for the superlet method: [superlet.py](https://github.com/esi-neuroscience/syncopy/blob/dev/syncopy/specest/superlet.py). So you could work with pure numpy/scipy/... and we can take care of the integration into SyNCoPy when building the time-lock submodule. Or what do you think @pantaray ?

**Comment by pantaray** [comment]
Good idea @tensionhead Sounds like a plan. Thank you for bringing this up, @KatharineShapcott !

**Comment by KatharineShapcott** [comment]
> So you could work with pure numpy/scipy/... and we can take care of the integration into SyNCoPy when building the time-lock submodule. 

Okay great thanks. That makes it easy, I'll try and write something, run a few tests and let you know if I still think it's suitable.

> I wonder what is the connection to the KS-Test, which also is parameter free and works on the cumulative distributions as well? 

They go into this in the methods, as far as I understood their jittering method seems to create a better null distribution https://elifesciences.org/articles/71969/figures#fig3s3

**Comment by tensionhead** [comment]
Ok great, we'll merge the latest submodule *connectivity* very very soon (hopefully tmrw) into our `dev` branch, then we'll create a dedicated `timelockanalysis` branch where you then could directly commit into. I will update this comment/ping you once we are there!

**Comment by tensionhead** [comment]
Ok @KatharineShapcott , with quite some delay we got everything in place.. I just created a new `timelockanalysis` branch!

**Comment by dfsp-spirit** [comment]
@KatharineShapcott Is this still relevant to you, and if so, do you still want to provide the core scipy/numpy code, as discussed above?
